R/randomForest_tidiers.R

```diff
@@ -2670,6 +2625,54 @@ glance.rpart <- function(x, pretty.name = FALSE, ...) {
   ret
 }
 
+# Builds partial_dependency object for rpart with same structure (a data.frame with attributes.) as edarf::partial_dependence.
+partial_dependence.rpart = function(fit, target, vars = colnames(data),
+  n = c(min(nrow(unique(data[, vars, drop = FALSE])), 25L), nrow(data)),
+  interaction = FALSE, uniform = TRUE, data, ...) {
+
+  predict.fun = function(object, newdata) {
+    # Within our use cases, rpart always returns numeric values or probability
+    # even with multiclass classification.
+    # In that case, just calling predict is enough.
+    # edarf code for ranger does something more for classification case,
+    # but it is not necessary here.
+    ret <- predict(object, newdata)
+    ret
+  }
+
+  args = list(
+    "data" = data,
+    "vars" = vars,
+    "n" = n,
+    "model" = fit,
+    "uniform" = uniform,
+    "predict.fun" = predict.fun,
+    ...
+  )
+  
+  if (length(vars) > 1L & !interaction) {
```

Comment:
Please add some comment for this condition.

Reply:
Added comment.